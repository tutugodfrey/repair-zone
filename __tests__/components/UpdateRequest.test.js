import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import ConnectedUpdateRequest, { UpdateRequest, mapStateToProps }
  from '../../client/src/components/UpdateRequest.jsx';

let unconnectedWrapper;
let connectedWrapper;
const requests = [{}, {}]
const requestToUpdate = {};
const serviceProviders = [];
const initialState = {
  requestReducer: {
    requests,
    requestToUpdate,
    serviceProviders,
  },
  pageReducer: {
    errorStatus: false,
    errorMessage: '',
  }
}
const mockStore = configureStore();
const store = mockStore(initialState);

describe('<UpdateRequest /> test', () => {
  describe('unconnected update request component', () => {
    beforeEach(() => {
      unconnectedWrapper = shallow(
      <UpdateRequest
        requests={requests}
        requestToUpdate={requestToUpdate}
        serviceProviders={serviceProviders}
      />);
    });
    test('should map the snapshot of the component', () => {
      expect(toJson(unconnectedWrapper)).toMatchSnapshot();
    });
  });
  describe('connected update request component', () => {
    beforeEach(() => {
      const user = {
        fullName: 'fullname',
        serviceName: 'servicename',
        isAdmin: true,
      }
      const requests = [
        {
          user,
          request: {
            id: 1,
            status: 'awaiting confirmation'
          }
        }
      ];
      store.getState().requestReducer.requests = requests;
      const dispatch = jest.fn();
      connectedWrapper = mount(
        <Provider store={store}>
          <ConnectedUpdateRequest
            dispatch={dispatch}
            serviceProviders={serviceProviders}
          />
        </Provider>
      );
    });
    test('should render a div with class card', () => {
      const divContainer = connectedWrapper.find('.card');
      expect(divContainer.length).toBe(1);
    })
    test('should render with an update button', () => {
      const renderedBtns = connectedWrapper.find('button');
      expect(renderedBtns.length).toBe(2)
    });
  });
  describe('mapStateToProps', () => {
    test('should map the state of the component correctly', () => {
      const appState = {
        requestReducer: {
          requests: [{}, {}, {}],
          requestToUpdate: {},
          serviceProviders: [],
        },
        pageReducer: {
          errorMessage: '',
          errorStatus: true,
        }
      }
      const expectedState = {
        requests: [{}, {}, {}],
        requestToUpdate: {},
        serviceProviders: [],
      }
      const componentState = mapStateToProps(appState);
      expect(componentState).toEqual(expectedState)
    })
  })
});
