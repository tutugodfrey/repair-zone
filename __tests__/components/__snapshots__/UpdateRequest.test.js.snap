// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateRequest /> test unconnected update request component should map the snapshot of the component 1`] = `
<div>
  <Connect(Modal) />
  <Div
    content={
      <Div
        content={
          <Form
            content={
              <div
                className="card"
              >
                <div
                  className="card-body"
                >
                  <h3
                    className="card-title text-center"
                  >
                    Make Repair Request
                  </h3>
                  <FormSelect
                    divClass=""
                    divId="service-div"
                    inputClass="form-control border-success"
                    inputId="admin"
                    inputName="adminId"
                    labelValue="Service Providers"
                    onChange={[Function]}
                    options={Array []}
                  />
                  <p>
                    Your previous selection is 
                    <strong />
                  </p>
                  <FormSelect
                    divClass=""
                    divId="category-div"
                    inputClass="form-control border-success"
                    inputId="category"
                    inputName="category"
                    labelValue="Category"
                    onChange={[Function]}
                    options={
                      Array [
                        "Select",
                        "Electrical",
                        "automobile",
                        "painting",
                        "capentary",
                        "electronics",
                      ]
                    }
                  />
                  <p>
                    Your previous selection is 
                    <strong />
                  </p>
                  <TextArea
                    cols="60"
                    divClass="form-group border-success"
                    id="description"
                    inputClass="form-control border-success"
                    labelValue="Description"
                    name="description"
                    onChange={[Function]}
                    placeholder="Describe you requirement here"
                    rows="10"
                    type="text"
                  />
                  <br />
                  <FormInput
                    divClass="form-group"
                    id="address"
                    inputClass="form-control border-success"
                    inputName="address"
                    labelValue="Address"
                    onChange={[Function]}
                    placeholder="address"
                    type="text"
                  />
                  <br />
                  <CheckBox
                    divClass="border-success"
                    divId="urgency-div"
                    inputClass="border-success"
                    inputId="urgent"
                    inputName="urgent"
                    labelClass="ml-3"
                    labelId="urgent-label"
                    labelValue="Urgent"
                    onChange={[Function]}
                    value=""
                  />
                  <Button
                    buttonClass="ml-1 my-2 px-5 py-2 btn btn-sm bg-success text-center text-white"
                    buttonId="user-undefined-update"
                    buttonName="Save Update"
                    onClick={[Function]}
                  />
                </div>
              </div>
            }
            formClass=""
            formId=""
          />
        }
        divClass="col-10 offset-1 col-sm-8 offset-sm-2 col-md-6 offset-md-3 py-5 my-3"
        divId=""
      />
    }
    divClass="row"
    divId=""
  />
</div>
`;
